File: spring-boot-oauth2-auth-server/src/main/java/com/bittokazi/oauth2/auth/server/app/repositories/master/TenantRepository.java
package com.bittokazi.oauth2.auth.server.app.repositories.master;

import com.bittokazi.oauth2.auth.server.app.models.master.Tenant;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface TenantRepository extends JpaRepository<Tenant, String> {

    Optional<Tenant> findOneByCompanyKey(String companyKey);

    Optional<Tenant> findOneByLogo(String logo);

    Optional<Tenant> findOneByDomain(String domain);

}

File: spring-boot-oauth2-auth-server/src/main/kotlin/com/bittokazi/oauth2/auth/server/app/repositories/master/TenantRepository.kt
package com.bittokazi.oauth2.auth.server.app.repositories.master

import com.bittokazi.oauth2.auth.server.app.models.master.Tenant
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface TenantRepository : JpaRepository<Tenant, String> {
    fun findOneByCompanyKey(companyKey: String): Optional<Tenant>

    fun findOneByLogo(logo: String): Optional<Tenant>

    fun findOneByDomain(domain: String): Optional<Tenant>
}
